### DISCUSSION TITLE
Skip Gradle initialization in the project

### DISCUSSION THREAD

**Discussion by ahus1** [JetBrains/Qodana#130] [Q&A]
For one of my projects I want to check the docs in the project using the Grazie and the AsciiDoc plugin. For that, I wouldn't need the Gradle parts of the project initialized when running Qodana: https://github.com/asciidoctor/asciidoctor-intellij-plugin/

Due to the initialization of Gradle, it downloads (unnecessarily) ~3,3 GB of data that it doesn't need. In addition, I keep wondering what Qodana uses as a cache key, that is: what change to wich file invalidates the cache -- see https://github.com/asciidoctor/asciidoctor-intellij-plugin/actions/runs/3268249646/jobs/5374431345 for a previous run.

Previously I ran Qodana only on the docs folder, but with the changes to inspect the latest Git commits this has proven no longer viable. 

So I wonder: Can I teach Qodana not to initialize Gradle? 

Thanks!

**Comment by brichbash** [comment]
Hello,

In order to skip the Gradle step you can either disable the plugin by passing `--property=idea.suppressed.plugins.id=com.intellij.gradle` to the action via `args` or just using non-jvm linter, for example, `qodana-js`. Grazie and your plugin should still work.

Considering the question about a  cache key, @tiulpin, could you please help?

**Comment by ahus1** [comment]
@brichbash - I now had a situation where no cache was found, and then the Qodana action printed

> Cache not found for input keys: qodana-refs/pull/1209/merge-269cc1d58159db037c52385c4041afd7756299a1, qodana-refs/pull/1209/merge-, qodana-

So I believe this is the answer why the cache is ever-increasing when I change dependencies. 

I applied the setting for the property, and it works well. I looked into the docs and found this also documented. The documentation also indicates that I can have multiple plugins disabled using a comma (`,`) ... unfortunately this doesn't work as expected with the action removes all commas from the `args` parameter as it calls `split(',')`

So (with a comma)

`--property=idea.suppressed.plugins.id=com.intellij.gradle,org.jetbrains.plugins.gradle`

ends up as (with a blank)

`--property=idea.suppressed.plugins.id=com.intellij.gradle org.jetbrains.plugins.gradle`

Is there a workaround for this?

**Reply by brichbash** [reply]
I created [an issue ](https://youtrack.jetbrains.com/issue/QD-4329/Its-not-possible-to-set-multiple-values-for-a-property-using-GitHub-Action )regarding the problem with passing multiple values to the property. However, in order to skip Gradle, disabling `com.intellij.gradle` should be enough. You missed a comma after `com.intellij.gradle` in [docs.yml](https://github.com/asciidoctor/asciidoctor-intellij-plugin/blob/main/.github/workflows/docs.yml):
```
args: >
            --linter,jetbrains/qodana-jvm-community:2022.2,
            --property=idea.suppressed.plugins.id=com.intellij.gradle
            -v,${{ github.workspace }}/grazie:/opt/idea/plugins/grazie,
```
So the plugin isn't disabled. Adding the comma should do it.
